pkg/api/steve/supportconfigs/supportconfigs.go

```diff
@@ -0,0 +1,57 @@
+package supportconfigs
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/version"
+)
+
+// supportConfig represents the serializable struct of support related information common to all rancher setups
+type supportConfig struct {
+	// SupportEligible - if this rancher product is eligible for support.
+	SupportEligible bool `json:"support_eligible,omitempty"`
+	// Platform - the platform (arm, x86_64, etc) rancher is running on. As of now, always x86
+	Platform string `json:"platform"`
+	// Product - suse product string identifying rancher - see getRancherProductString
+	Product string `json:"product"`
```

Comment:
Do we need these https://github.com/rancher/csp-adapter/blob/d8ef954b0e111203872bec9cf5dc1c5e09d59c93/pkg/supportconfig/generator.go#L32-L38 ?

Reply:
No. These values are provided by the CSP adapter. Rancher lacks the credentials to gather the information on which aws account this is registered to and it lacks the logic to calculate if it is in compliance with the licenses purchased. 

In short - these values are produced by the csp adapter.

Reply:
In that case it seems like there is a bit of duplication between the Rancher side of this and the CSP adapter side, given that they are using almost-but-not-quite-the-same types. It would be nice if that could be reduced somewhat.

Reply:
From the perspective of the code, these aren't really using the same types. I've intentionally made this use a map-based merge rather than using a well-defined struct so that the CSP adapter can supply whatever values that it feels are appropriate. 

The csp adapter does already need to import rancher (to create the user notifications), so in theory I could use rancher as the source of what fields are available from an overall perspective. However, I would want to avoid this since it would require changes in rancher's side for new fields that the csp adapter feels are appropriate.

For example, let's say that I port over the entire type that used to be in the csp adapter into core rancher. If the adapter needs to supply a new field (say something specific to the aws license like "Number of Licenses purchased"), I would first need to add the field in rancher, then update the adapter's dependencies, and then I could supply the new field. That seems like a lot of extra work.

The Design I was going for here was for Rancher to define a few things itself and then include whatever arbitrary values the csp adapter supplied. I know that this isn't the clearest model, but it is more flexible and allows the adapter to decide what fields are relevant, rather than requiring rancher to be an overriding source of truth that the adapter relies upon.

Reply:
Okay, this makes more sense in context